src/Main.cpp

```diff
@@ -144,29 +145,32 @@ class MessageSenderTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
 
     void task() override {
         if (m_stream != NULL) {
-            HiveMindHostSerializer serializer(*m_stream);
+            SerializerType serializer(*m_stream);
             MessageSender messageSender(m_streamQueue, serializer, BSPContainer::getBSP(),
                                         m_logger);
-            while (m_stream->isConnected()) {
-                // Verify that we have a message to process
-                if (m_streamQueue.isEmpty()) {
-                    m_streamQueue.wait(500);
-                }
-                if (!messageSender.processAndSerialize()) {
-                    m_logger.log(LogLevel::Warn, "Fail to process/serialize in %s", m_taskName);
+            while (true) {
+                if (m_stream->isConnected()) {
+                    // Verify that we have a message to process
+                    if (m_streamQueue.isEmpty()) {
+                        m_streamQueue.wait(500);
+                    }
+                    if (!messageSender.processAndSerialize()) {
+                        m_logger.log(LogLevel::Warn, "Fail to process/serialize in %s", m_taskName);
+                    }
                 }
             }
         }
     }
 };
 
+template <typename SerializerType>
```

Comment:
You could the default template param serializer to a HiveMindHostSerializer, avoiding the need to specify it for the other tasks.